extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/EnvVarHolder.java

```diff
@@ -0,0 +1,68 @@
+package io.quarkus.kubernetes.deployment;
+
+import java.util.Collection;
+import java.util.Map;
+
+import io.quarkus.kubernetes.spi.KubernetesEnvBuildItem;
+
+import static io.quarkus.kubernetes.spi.KubernetesEnvBuildItem.EnvType.configmap;
+import static io.quarkus.kubernetes.spi.KubernetesEnvBuildItem.EnvType.field;
+import static io.quarkus.kubernetes.spi.KubernetesEnvBuildItem.EnvType.secret;
+import static io.quarkus.kubernetes.spi.KubernetesEnvBuildItem.EnvType.var;
+
+/**
+ * Common interface for configuration entities holding environment variables meant to be injected into containers.
+ */
+public interface EnvVarHolder {
+    /**
+     * Retrieves the definition of environment variables to add to the application's container.
+     *
+     * @return the associated {@link EnvVarsConfig} holding the definition of which environment variables to add
+     */
+    EnvVarsConfig getEnv();
+    
+    /**
+     * @deprecated use {@link #getEnv()} instead
+     */
+    @Deprecated
+    Map<String, EnvConfig> getEnvVars();
+    
+    /**
+     * Specifies which platform this EnvVarHolder targets. This needs to be overridden by implementations that care about which
+     * platform they target.
+     *
+     * @return the name of the targeted platform e.g. {@code kubernetes}
+     */
+    default String getTarget() {
+        return "UNUSED";
```

Comment:
This feels weird, as we already have `deployment target` which may return one or more target platforms, can we use that fields instead?

Reply:
Not sure I follow here. I might have missed something. Could you elaborate, please?

Reply:
IIUC, @iocanel is wondering if `KubernetesConfig#deploymentTarget` can be used instead

Reply:
I will check but I don't think it's feasible because that field is used to define which of the implementations of `EnvVarHolder` need to be created (e.g. KubernetesConfig) whereas `getTarget` lets the generic code know which platform a given implementation targets. I agree that this is a little weird but it is convenient to have common processing of env vars for all platforms (including `ContainerConfig` which actually doesn't use the target field, hence the default implementation).

Reply:
If I understand correctly, the method is used the specify the 'kind' of the config that is bearing the envars, right? If this is the case, I feel that there shouldn't be a default value returned and maybe the value could be derived from the classname. In any case I feel we need a different name for this.

Reply:
The value returned by the `getTarget` method is used to create `KubernetesEnvBuildItem` and the value is then used by the `KubernetesProcessor` to decorate dekorate resources. Of note, the value is actually unused for the `ContainerConfig` implementation since it is not consumed by the `KubernetesProcessor`.
I guess we could derive it from the implementing class name but that seems dangerous to me since that value is used by dekorate to generate the file names. Also, we could certainly change the name but I simply reused the name that already existed on `KubernetesEnvBuildItem` which this method is used to create.